Shopify analytics 2 (#1325)
Shopify/hydrogen: docs/components/framework/shopifyanalytics.md

## Configuration

To set up the `ShopifyAnalytics` component in your Hydrogen app, add the `ShopifyAnalyticsServerConnector` property to your [Hydrogen configuration file](https://shopify.dev/custom-storefronts/hydrogen/framework/hydrogen-config):

{% codeblock file, filename: 'hydrogen.config.js' %}

```jsx
import {
  ...
  ShopifyAnalyticsServerConnector,
} from '@shopify/hydrogen';
...
export default defineConfig({
  ...
  serverAnalyticsConnectors: [
    PerformanceMetricsServerAnalyticsConnector,
    // The `serverAnalyticsConnectors` property allows you to send analytics data from the server in your Hydrogen app.
    ShopifyAnalyticsServerConnector,
  ],
});
```

{% endcodeblock %}

After you've updated your Hydrogen configuration file, add the `ShopifyAnalytics` component in `App.server.jsx`.

{% codeblock file, filename: 'App.server.jsx' %}

```jsx
function App() {
  return (
    <Suspense fallback={<LoadingFallback />}>
      <ShopifyProvider>
        ...
        <ShopifyAnalytics />
      </ShopifyProvider>
    </Suspense>
  );
}
```

{% endcodeblock %}

> If you have a custom domain or you are using sub domains, you can set the cookie domain of

Reviewer: ```suggestion
If you have a custom domain or you're using sub-domains, then you can set the cookie domain of
```

the Shopify analytics component so that cookies persists for your root domain.

> ```jsx
> <ShopifyAnalytics cookieDomain="my-shop.com" />
> ```

Reviewer: @wizardlyhel Would this also be set up in `App.server.jsx`? Adding the filename might be helpful as a reference point for devs:

````suggestion
{% codeblock file, filename: 'App.server.jsx' %}

```jsx
<ShopifyAnalytics cookieDomain="my-shop.com" />
```

{% endcodeblock %}
````

> Otherwise, this component will use the `storeDomain` value in the `hydrogen.config.js`
> as the default cookie domain or leave it blank when the specified cookie domain does
> not match `window.location.hostname`.

Reviewer: ```suggestion
If you're not using custom domains or sub-domains, then the `ShopifyAnalytics` component uses the `storeDomain` value in the Hydrogen configuration file as the default cookie domain or leaves it blank when the specified cookie domain doesn't match `window.location.hostname`.
```

> ### Hydrogen analytics connect with checkout analytics

Reviewer: ```suggestion
### Connecting Hydrogen analytics with Shopify checkout
```

There is a strict requirement for the analytic cookies to be set at the 1st party
domain. This means that when a buyer navigates from your Hydrogen storefront to
Shopify checkout, the domain name must stay the same.

> You can acheive this by assigning a subdomain to your online store. For example:

Reviewer: ```suggestion
You can achieve this by assigning a sub-domain to your online store. For example, you can do the following tasks:
```

> * Set your Hydrogen store domain at `https://www.my-awesome-hydrogen-store.com`
> * Attach a new subdomain to your online store at `https://checkout.my-awesome-hydrogen-store.com`
> * Set the cookieDomain to the same root domain `<ShopifyAnalytics cookieDomain="my-awesome-hydrogen-store.com" />`

Reviewer: ```suggestion
- Set your Hydrogen store domain to `https://www.my-awesome-hydrogen-store.com`.
- Attach a new sub-domain to your online store at `https://checkout.my-awesome-hydrogen-store.com`.
- Set the `cookieDomain` to the same root domain: `<ShopifyAnalytics cookieDomain="my-awesome-hydrogen-store.com" />`
```

> > Note: It is expected behaviour that Hydrogen analytics and checkout analytics do not connect
> when in development and preview mode. This will only connect in production mode.

Reviewer: ```suggestion
> Note: 
> Hydrogen analytics and Shopify checkout can only be connected in production. They can't be connected in development and preview modes. 
```

The `ShopifyAnalytics` component sends commerce-related analytics to Shopify. By adding the `ShopifyAnalytics` component to your Hydrogen app, you can view key sales, orders, and online store visitor data from the [Analytics dashboard in your Shopify admin](https://help.shopify.com/en/manual/reports-and-analytics/shopify-reports/overview-dashboard).

> > Note: This is a still a work in progress. Only Online Store page view and session related analytic reports are working at this moment.

Author: sounds good

{% codeblock file, filename: 'hydrogen.config.js' %}

```jsx
import {
  ...
  ShopifyAnalyticsServerConnector,
} from '@shopify/hydrogen';
...
export default defineConfig({
  ...
  serverAnalyticsConnectors: [
    PerformanceMetricsServerAnalyticsConnector,
    // The `serverAnalyticsConnectors` property allows you to send analytics data from the server in your Hydrogen app.
    ShopifyAnalyticsServerConnector,
  ],
});
```

{% endcodeblock %}

After you've updated your Hydrogen configuration file, add the `ShopifyAnalytics` component in `App.server.jsx`.

{% codeblock file, filename: 'App.server.jsx' %}

```jsx
function App() {
  return (
    <Suspense fallback={<LoadingFallback />}>
      <ShopifyProvider>
        ...
        <ShopifyAnalytics />
      </ShopifyProvider>
    </Suspense>
  );
}
```

{% endcodeblock %}

If you have a custom domain or you're using sub-domains, then you can set the cookie domain of
the `ShopifyAnalytics` component so that cookies persists for your root domain:

{% codeblock file, filename: 'App.server.jsx' %}

```jsx

> <ShopifyAnalytics cookieDomain="my-shop.com" />

Author: No
* maxage: Defined by Shopify to be 2 years for `_shopify_y` and 30 mins for `_shopify_s
* path: should always be `/`

> <ShopifyAnalytics cookieDomain="my-shop.com" />

Author: This might change when CHIP is fully supported
